modules/assignment/src/main/java/com/intuit/wasabi/assignment/impl/AssignmentsImpl.java

```diff
@@ -609,6 +610,15 @@ protected Assignment getAssignment(User.ID userID, Application.Name applicationN
             if (createAssignment) {
                 if (experiment.getState() == Experiment.State.PAUSED) {
                     return nullAssignment(userID, applicationName, experimentID, Assignment.Status.EXPERIMENT_PAUSED);
+                } else if (experiment.getIsRapidExperiment() != null && experiment.getIsRapidExperiment()) {
+                    //Get the latest state from the DB for rapid experiments if they are not in stopped state.
+                    Experiment rapidExperiment = experimentUtil.getExperiment(experiment.getID());
+                    Experiment.Label rapidExperimentLabel = null != rapidExperiment ? rapidExperiment.getLabel(): null;
```

Comment:
Mind wrapping the conditional in (...) to make the assignment more readable? E.g.:

Experiment.Label rapidExperimentLabel = (rapidExperiment != null ? rapidExperiment.getLabel() : null);